### PR TITLE
docs: add documentation for cli options

### PR DESCRIPTION
**Pull request by deepeshaburse**
### Description
To fix issue: https://github.com/conda/conda/issues/11368

To add documentation for:
- [ ] NoRegistry
- [ ] NoShortcuts
- [ ] NoScripts
- [ ] RegisterPython
- [ ] CheckPathLength 

### Checklist - did you ...

- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by deepeshaburse** [comment]
Hi @jaimergp, kindly check this PR out. Thank you!

**Comment by jaimergp** [comment]
Worth a look to make sure there are no overlaps: https://conda-dot-org.netlify.app/community#the-many-meanings-of-conda

**Review by jaimergp — CHANGES_REQUESTED** [review]
Thanks for starting this!

This is the overall structure I imagine for this section:

- H1: Command-line options of constructor-generated installers
- H2: Shell-based installers for Linux and macOS
  - Describe the `.sh` CLI ([described in this block](https://github.com/conda/constructor/blob/da3eec21f1f62fac88ebbae598aebcccb2c6f829/constructor/header.sh#L54-L75))
- H2: Windows installers
  - Include the flags already in this PR
  - Add the [ones that are included in all NSIS installers](https://nsis.sourceforge.io/Docs/Chapter3.html#installerusage) (not just constructor ones)
  - Note the default NSIS parametes need to be _last_, i.e. after the constructor ones. The `/D` parameter should be the last of those, and the value unquoted. Yea, I know 🤷 
  - Cover the uninstaller CLI too (same page as item above)
- H2: PKG installers for macOS
  - Almost no CLI. The few options available are [discussed in this script](https://github.com/conda/constructor/blob/da3eec21f1f62fac88ebbae598aebcccb2c6f829/scripts/run_examples.py#L166-L167). [Refer to Apple's `installer` docs too](https://ss64.com/osx/installer.html) for extra details (no need to copy here, just a link suffices).


---

Overall I feel that the style of the output should be either a table, or an itemized list, but I wouldn't use headers for each CLI option as we are doing now. So something like this:

```markdown
# CLI options for constructor-generated installers
...
## Windows installers

Windows installers have the following CLI options available:

- `/NoRegistry`: ...

After these flags, you can also add any NSIS standard flags:

- `/S`: silent mode
- `/D`: target directory. Do not quote the value even if it has spaces! This needs to be the last option added in the CLI.

Note NSIS installers won't add any output to the terminal. We recommend running them in one of the following ways:

- With  CMD, use ...
- With Powershell, use ...

```

I hope that's clear!

**Comment by deepeshaburse** [comment]
> Thanks for starting this!
> 
> 
> 
> This is the overall structure I imagine for this section:
> 
> 
> 
> - H1: Command-line options of constructor-generated installers
> 
> - H2: Shell-based installers for Linux and macOS
> 
>   - Describe the `.sh` CLI ([described in this block](https://github.com/conda/constructor/blob/da3eec21f1f62fac88ebbae598aebcccb2c6f829/constructor/header.sh#L54-L75))
> 
> - H2: Windows installers
> 
>   - Include the flags already in this PR
> 
>   - Add the [ones that are included in all NSIS installers](https://nsis.sourceforge.io/Docs/Chapter3.html#installerusage) (not just constructor ones)
> 
>   - Note the default NSIS parametes need to be _last_, i.e. after the constructor ones. The `/D` parameter should be the last of those, and the value unquoted. Yea, I know 🤷 
> 
>   - Cover the uninstaller CLI too (same page as item above)
> 
> - H2: PKG installers for macOS
> 
>   - Almost no CLI. The few options available are [discussed in this script](https://github.com/conda/constructor/blob/da3eec21f1f62fac88ebbae598aebcccb2c6f829/scripts/run_examples.py#L166-L167). [Refer to Apple's `installer` docs too](https://ss64.com/osx/installer.html) for extra details (no need to copy here, just a link suffices).
> 
> 
> 
> 
> 
> ---
> 
> 
> 
> Overall I feel that the style of the output should be either a table, or an itemized list, but I wouldn't use headers for each CLI option as we are doing now. So something like this:
> 
> 
> 
> ```markdown
> 
> # CLI options for constructor-generated installers
> 
> ...
> 
> ## Windows installers
> 
> 
> 
> Windows installers have the following CLI options available:
> 
> 
> 
> - `/NoRegistry`: ...
> 
> 
> 
> After these flags, you can also add any NSIS standard flags:
> 
> 
> 
> - `/S`: silent mode
> 
> - `/D`: target directory. Do not quote the value even if it has spaces! This needs to be the last option added in the CLI.
> 
> 
> 
> Note NSIS installers won't add any output to the terminal. We recommend running them in one of the following ways:
> 
> 
> 
> - With  CMD, use ...
> 
> - With Powershell, use ...
> 
> 
> 
> ```
> 
> 
> 
> I hope that's clear!

Sure, thank you for such a detailed explanation. I'll make those changes shortly.

**Comment by deepeshaburse** [comment]
@jaimergp I have updated the structure of the documentation and I should be able to add some content by tomorrow. Thanks!

File: docs/source/clioptions.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+We have the following CLI options available for MacOS and Linux:
+
+- If the installer is running in a batch, `-i` is used to run it in an interactive mode. Otherwise, the installer runs in batch mode without manual intervention by `-b`. 
+- `f` returns no error is the prefix already exists.
+- `h` allows us to display this help message and exit.
+- You may use `-k` to keep the package cache after installation.
+- You can use `-p PREFIX` to install a prefix. Kindly note that the path should not contain any spaces.
+- If you wish to skip running `pre/post-link/install`scripts, you may use `-s`.
+- `-u` is used to update an existing installation.
+- Considering you have conda installed, `-t` can be used to run package tests after installation (it may install conda-build).
```
Comment (jaimergp) [inline docs/source/clioptions.md:14]: I'd rather have the options listed as we usually find them in help messages:

- Hyphenated flags first, then description
- Order is usually alphabetic, but feel free to group them differently if it makes sense.

Comment (deepeshaburse) [reply docs/source/clioptions.md:14]: Sure, I'll make those changes. I decided on the order based on how they were defined in the code. I'll try to put it alphabetically, if it doesn't work out, I'll keep the order the same until the next review.

**Comment by deepeshaburse** [comment]
@jaimergp I have updated the documentation according to your suggestions, kindly give it a look!

**Review by jaimergp — APPROVED** [review]
Thank you so much @deepeshaburse for compiling all this info! I added a couple changes to your PR to push it over the finish line!

**Pull request merged**